packages/serialize/types/index.d.ts

```diff
@@ -7,18 +7,19 @@ import * as CSS from 'csstype'
 export { RegisteredCache, SerializedStyles }
 
 export type CSSProperties = CSS.PropertiesFallback<number | string>
-export type CSSPropertiesWithMultiValues = {
+export type CSSPropertiesWithMultiValues<Props> = {
   [K in keyof CSSProperties]:
     | CSSProperties[K]
     | ReadonlyArray<Extract<CSSProperties[K], string>>
+    | ((props: Props) => number | string)
```

Comment:
Those 2 seem to be inconsistent. Why do we allow `number | string` to be returned from a function but the array case does only allow `string`?

Reply:
If my understanding is correct, the `string` in `ReadonlyArray<Extract<CSSProperties[K], string>>` is to pull possible values from the `CSSProperties[K]`.

For callback as a value, I need to use `number | string` to handle this case:

```js
styled('div')({
  lineHeight: props => props.lineHeight || 1.2, // number is a valid unit for some CSS properties
})
```

Reply:
But isn't this valid then?
```ts
styled('div')({
  lineHeight: [1.2],
})
```

Note that array values here are somewhat weird. They mainly exist so you can do stuff like this:
```ts
// display: -webkit-flex;
// display: flex;
styled('div')({
  display: ['-webkit-flex', 'flex'],
})
```

We can't support this otherwise as we can't observe duplicate properties at runtime. Perhaps this particular use case actually means that only string values should be supported here as numerical values used like this don't make a lot of sense (unless dunno, maybe mixing a numerical fallback with some keyword~ would make sense?)

That also kinda means that we should support this at type level:
```ts
styled('div')({
  display: (props) => ['-webkit-flex', 'flex'],
})
```

Reply:
👍 Fixed in https://github.com/emotion-js/emotion/pull/3146/commits/cc913db69c1405a915a982d9e86b80f44c50f9ee